### PR TITLE
Don't use '-' as a delimiter in BigQuerySinkMetrics

### PR DESCRIPTION
**Pull request by JayajP**
`BigQuerySinkMetrics` support labelled metrics by concatenating a base metric name with the metric labels. The characaters `-`, `:` and `;` are used as delimiters in this process. Using `-` as a delimiter causes issues since `-` can also be used a BQ table name. This PR replaces the `-` delimiter with `*` which cannot be used in table ids. (All other metric labels are generated from known enums so we do not need to worry about them).

Additionally changes:
- In `BigQuerySinkMetrics::ParsedMetricName::parsemetricname` create string splitters with `limit(2)` when we split the base metric name from labels, and when we split a metric labels keys and values. This makes parsing more resilient incase delimiters are added to metric labels. 
- modify `BigQuerySinkMetrics::ParsedMetricName::parsemetricname` to return an empty `Optional` instead of null when parsing fails.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by JayajP** [comment]
R: @m-trieu

File: runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/MetricsToPerStepNamespaceMetricsConverter.java [hunk 1]
```diff
@@ -52,16 +52,16 @@ private static Optional<MetricValue> convertCounterToMetricValue(
       return Optional.empty();
     }
 
-    BigQuerySinkMetrics.ParsedMetricName labeledName =
+    Optional<BigQuerySinkMetrics.ParsedMetricName> labeledName =
```
Comment (m-trieu) [inline runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/MetricsToPerStepNamespaceMetricsConverter.java:55]: nit: you can do 

```
return BigQuerySinkMetrics.parseMetricName(metricName.getName())
// filter for basenames !empty 
.filter(labeledName -> !labeledName.isEmpty())
// Only called if above predicate is true
.map(labeledName -> new MetricValue()
            .setMetric(labeledName.getBaseName())
            .setMetricLabels(labeledName.getMetricLabels())
            .setValueInt64(value)));
```

if `BigQuerySinkMetrics.parseMetricName(metricName.getName())` returns null or the filter predicate fails will default to Optional.empty().

this is more of a style thing

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySinkMetrics.java [hunk 2]
```diff
@@ -40,10 +41,10 @@
  *
  * <p>In general metrics be in the namespace 'BigQuerySink' and have their name formatted as:
  *
- * <p>'{baseName}-{metricLabelKey1}:{metricLabelVal1};...{metricLabelKeyN}:{metricLabelValN};'
+ * <p>'{baseName}*{metricLabelKey1}:{metricLabelVal1};...{metricLabelKeyN}:{metricLabelValN};'
  */
 public class BigQuerySinkMetrics {
-  private static Boolean supportMetricsDeletion = false;
+  private static boolean supportMetricsDeletion = false;
```
Comment (m-trieu) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySinkMetrics.java:47]: if this is a constant should be styled as 
`private static final boolean SUPPORT_METRICS_DELETION = false;`

**Review by m-trieu — APPROVED** [review]
Other than style comments LGTM

File: runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/MetricsToPerStepNamespaceMetricsConverter.java (hunk 1 above)
Comment (JayajP) [reply runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/MetricsToPerStepNamespaceMetricsConverter.java]: Done.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySinkMetrics.java (hunk 2 above)
Comment (JayajP) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySinkMetrics.java:47]: This value is mutable

**Pull request merged**